### PR TITLE
Unify some code in DrawEngine

### PR DESCRIPTION
**Pull request by hrydgard**

### CODE REVIEW

File: GPU/Common/DrawEngineCommon.cpp [hunk 1]
```diff
@@ -516,3 +517,48 @@ void DrawEngineCommon::DecodeVertsStep(u8 *dest, int &i, int &decodedVerts) {
 		i = lastMatch;
 	}
 }
+
+
```
Comment (unknownbrackets) [inline GPU/Common/DrawEngineCommon.cpp]: nit: Spacious.

-[Unknown]

File: GPU/Common/IndexGenerator.cpp [hunk 2]
```diff
@@ -212,8 +212,8 @@ void IndexGenerator::TranslateLineStrip(int numInds, const ITypeLE *inds, int in
 template <class ITypeLE, int flag>
 void IndexGenerator::TranslateList(int numInds, const ITypeLE *inds, int indexOffset) {
 	indexOffset = index_ - indexOffset;
-	if (indexOffset == 0) {
-		memcpy(inds_, inds, numInds);
+	if (sizeof(ITypeLE) == sizeof(inds[0]) && indexOffset == 0) {
```
Comment (unknownbrackets) [inline GPU/Common/IndexGenerator.cpp]: Hmm, isn't that by definition `const ITypeLE *inds`?  Should be `inds_`.

-[Unknown]

**Review by unknownbrackets — CHANGES_REQUESTED** [review]

File: GPU/Common/IndexGenerator.cpp (hunk 2 above)
Comment (hrydgard) [inline GPU/Common/IndexGenerator.cpp]: Eh, yes :) Thanks.

**Comment by hrydgard** [comment]
Fixed the commented things, rebased and merged.

**Pull request merged**